validator/main.py

```diff
@@ -97,47 +107,101 @@ def __init__(
         else:
             self._invalid_topics = invalid_topics
 
-        self._device = to_int(device)
+        self._device = (
+            str(device).lower()
+            if str(device).lower() in ["cpu", "mps"]
+            else int(device)
+        )
         self._model = model
         self._disable_classifier = disable_classifier
         self._disable_llm = disable_llm
+        self._classifier_api_endpoint = classifier_api_endpoint
 
-        if not model_threshold:
-            model_threshold = 0.5
-        else:
-            self._model_threshold = model_threshold
+        self._zero_shot_threshold = zero_shot_threshold
+        if self._zero_shot_threshold < 0 or self._zero_shot_threshold > 1:
+            raise ValueError("zero_shot_threshold must be a number between 0 and 1")
 
+        self._llm_threshold = llm_theshold
+        if self._llm_threshold < 0 or self._llm_threshold > 5:
+            raise ValueError("llm_threshold must be a number between 0 and 5")
         self.set_callable(llm_callable)
 
-    def get_topic_ensemble(
-        self, text: str, candidate_topics: List[str]
-    ) -> ValidationResult:
-        topic, confidence = self.get_topic_zero_shot(text, candidate_topics)
-
-        if confidence > self._model_threshold:
-            return self.verify_topic(topic)
+        if self._classifier_api_endpoint is None:
+            self._classifier = pipeline(
+                "zero-shot-classification",
+                model=self._model,
+                device=self._device,
+                hypothesis_template="This example has to do with topic {}.",
+                multi_label=True,
+            )
         else:
-            return self.get_topic_llm(text, candidate_topics)
+            # TODO api endpoint
```

Comment:
Will you address this TODO in another PR?

Reply:
Yes, there is actually a PR out for this for a basic implementation. However we are scoping out a more permanent solution that will hopefully make this cleaner. Still a WIP